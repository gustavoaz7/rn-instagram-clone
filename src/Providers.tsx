import React, { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components/native';
import { Provider } from 'react-redux';
import { theme as stylesTheme } from './styles/theme';
import { store as reduxStore } from './redux/store';

type TProvidersProps = {
  children: ReactNode;
  store?: typeof reduxStore;
  theme?: typeof stylesTheme;
};

export function Providers({
  children,
  store = reduxStore,
  theme = stylesTheme,
}: TProvidersProps): JSX.Element {
  return (
    <Provider store={store}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </Provider>
  );
}
