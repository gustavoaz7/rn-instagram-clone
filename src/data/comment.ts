import { TComment } from '../types';

export const generateComment = (): TComment => ({
  id: `${Math.random()}`,
  owner: {
    id: `${Math.random()}`,
    profilePicUrl: 'https://cdn.fakercloud.com/avatars/antonyzotov_128.jpg',
    username: 'Vivienne_Doyle',
  },
  createdAt: 1620488304326,
  text: 'Aspernatur quod ea unde repellendus magni nihil sunt.',
  likedBy: ['Frankie30', 'Patricia24', 'Westley_Ebert49'],
});
