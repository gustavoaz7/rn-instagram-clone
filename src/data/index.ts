import type {
  TLike,
  TComment,
  TPostMedia,
  TPost,
  TOwner,
  TStory,
  TStoryMedia,
  TProfile,
  TTappableObject,
} from '../types';

export const generateMockOwner = (): TOwner => ({
  id: `${Math.random()}`,
  profilePicUrl: 'https://avatars.githubusercontent.com/u/28453820?v=4',
  username: 'gustavoaz7',
});

export const generateMockLike = (): TLike => ({
  id: `${Math.random()}`,
  owner: generateMockOwner(),
  createdAt: 1620488304326,
});

export const generateMockComment = (likesQty = 1): TComment => ({
  id: `${Math.random()}`,
  owner: {
    id: `${Math.random()}`,
    profilePicUrl: 'https://cdn.fakercloud.com/avatars/antonyzotov_128.jpg',
    username: 'Vivienne_Doyle',
  },
  createdAt: 1620488304326,
  text: 'Aspernatur quod ea unde repellendus magni nihil sunt.',
  previewLikes: {
    count: likesQty,
    likes: [...Array(likesQty)].map(generateMockLike),
  },
  viewerHasLiked: false,
});

export const generateMockTappableObject = ({
  type = 'mention',
  text = 'random',
  x = 0.1,
  y = 0.2,
  scale = 1.1,
  rotation = 300,
}: Partial<TTappableObject> = {}): TTappableObject => ({
  id: `${Math.random()}`,
  type,
  text,
  x,
  y,
  scale,
  rotation,
});

export const generateMockPostMedia = ({ index = 0 } = {}): TPostMedia => ({
  id: `${Math.random()}`,
  url: `http://placeimg.com/640/480/cats?${index}`,
  owner: generateMockOwner(),
  tappableObjects: [],
});

export const generateMockPost = ({
  mediasQty = 1,
  likesQty = 3,
  commentsQty = 1,
} = {}): TPost => ({
  id: `${Math.random()}`,
  owner: generateMockOwner(),
  createdAt: 1620488304326,
  medias: [...Array(mediasQty)].map(generateMockPostMedia),
  caption:
    'Tempora rerum minima dolor quibusdam blanditiis ea et. Earum iste iure expedita velit laudantium inventore quia qui. Distinctio consectetur accusamus voluptatum hic rem at quas. Incidunt ipsum sed illo molestiae vitae ullam sit consequatur autem. Ut voluptatem nisi. Velit ex dignissimos sed aut.',
  previewLikes: {
    count: likesQty,
    likes: [...Array(likesQty)].map(generateMockLike),
  },
  previewComments: {
    count: commentsQty,
    comments: [...Array(commentsQty)].map(generateMockComment),
  },
  location: 'Daronville',
  viewerHasLiked: false,
});

export const generateMockStoryMedia = ({
  viewersQty = 0,
  index = 0,
} = {}): TStoryMedia => ({
  id: `${Math.random()}`,
  url: `http://placeimg.com/640/480/cats?${index}`,
  owner: generateMockOwner(),
  tappableObjects: [],
  previewViewers: {
    count: viewersQty,
    viewers: [...Array(viewersQty)].map(generateMockOwner),
  },
  takenAt: 1620488304326,
  expiresAt: 1620488304326 + 3600 * 24,
});

export const generateMockStory = ({
  mediasQty = 3,
  viewersQty = 0,
} = {}): TStory => ({
  id: `${Math.random()}`,
  owner: generateMockOwner(),
  expiresAt: 0,
  latestMediaAt: 0,
  seenAt: 0,
  medias: [...Array(mediasQty)].map((_, index) =>
    generateMockStoryMedia({ viewersQty, index }),
  ),
});

export const generateMockProfile = (): TProfile => ({
  bio: 'animi eum earum aut explicabo',
  followCount: 47,
  followedByCount: 50,
  followedByViewer: false,
  followsViewer: false,
  fullName: 'Gustavo Az',
  postsCount: 20,
  profilePicUrl: 'https://avatars.githubusercontent.com/u/28453820?v=4',
  username: 'gustavoaz7',
  story: generateMockStory(),
});
