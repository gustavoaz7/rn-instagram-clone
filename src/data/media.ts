import type { TPostMedia } from '../types';

export const generatePostMedia = (): TPostMedia => ({
  id: `${Math.random()}`,
  url: 'http://placeimg.com/640/480/cats',
  owner: {
    id: `${Math.random()}`,
    profilePicUrl: 'https://avatars.githubusercontent.com/u/28453820?v=4',
    username: 'gustavoaz7',
  },
  tappableObjects: [],
});
