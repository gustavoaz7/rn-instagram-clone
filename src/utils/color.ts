const HEX_COLOR_RE = /^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/;

export function addAlphaToHEX(hex: string, alpha: number) {
  if (!HEX_COLOR_RE.test(hex) || alpha < 0 || alpha > 1) return hex;

  let cleanHex = hex;
  if (hex.length === 4) {
    cleanHex = `#${hex[1].repeat(2)}${hex[2].repeat(2)}${hex[3].repeat(2)}`;
  }

  return `${cleanHex}${Math.round(255 * alpha).toString(16)}`;
}
