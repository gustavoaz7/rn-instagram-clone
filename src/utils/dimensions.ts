import { Dimensions } from 'react-native';

export const SCREEN_WIDTH = Dimensions.get('screen').width;
