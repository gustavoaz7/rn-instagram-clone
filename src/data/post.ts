import { TPost } from '../types';
import { generatePostMedia } from './media';

export const createMockPost = (): TPost => ({
  id: `${Math.random()}`,
  owner: {
    id: `${Math.random()}`,
    profilePicUrl: 'https://avatars.githubusercontent.com/u/28453820?v=4',
    username: 'gustavoaz7',
  },
  createdAt: 1620488304326,
  medias: [generatePostMedia()],
  caption:
    'Tempora rerum minima dolor quibusdam blanditiis ea et. Earum iste iure expedita velit laudantium inventore quia qui. Distinctio consectetur accusamus voluptatum hic rem at quas. Incidunt ipsum sed illo molestiae vitae ullam sit consequatur autem. Ut voluptatem nisi. Velit ex dignissimos sed aut.',
  likedBy: ['Tito.Zieme', 'Cleo.Runte69', 'Laverna.Emmerich24'],
  comments: [
    {
      id: `${Math.random()}`,
      owner: {
        id: `${Math.random()}`,
        profilePicUrl: 'https://cdn.fakercloud.com/avatars/antonyzotov_128.jpg',
        username: 'Vivienne_Doyle',
      },
      createdAt: 1620488304326,
      text: 'Aspernatur quod ea unde repellendus magni nihil sunt.',
      likedBy: ['Frankie30', 'Patricia24', 'Westley_Ebert49'],
    },
  ],
  location: 'Daronville',
});
