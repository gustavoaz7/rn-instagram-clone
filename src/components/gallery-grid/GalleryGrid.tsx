import React, { memo, useCallback, useRef } from 'react';
import { FlatList, ListRenderItem, StyleProp, ViewStyle } from 'react-native';
import styled from 'styled-components/native';
import { TPost } from '../../types';
import { SCREEN_WIDTH } from '../../utils/dimensions';
import LayersSvg from '../../../assets/svg/layers.svg';

export type TGalleryGridProps = {
  posts: TPost[];
  columns?: number;
  gap?: number;
};

export function GalleryGrid({
  posts,
  columns = 3,
  gap = 2,
}: TGalleryGridProps): JSX.Element {
  const contentContainerStyle = useRef<StyleProp<ViewStyle>>({
    width: SCREEN_WIDTH,
  }).current;
  const columnWrapperStyle = useRef<StyleProp<ViewStyle>>({
    paddingTop: gap,
    justifyContent: 'space-between',
  }).current;

  const renderItem = useCallback<ListRenderItem<TPost>>(
    ({ item: post }) => (
      <GalleryGrid.Item key={post.id} post={post} columns={columns} gap={gap} />
    ),
    [columns, gap],
  );

  return (
    <FlatList
      data={posts}
      renderItem={renderItem}
      numColumns={columns}
      columnWrapperStyle={columnWrapperStyle}
      contentContainerStyle={contentContainerStyle}
    />
  );
}

type TGaleryGridItemProps = {
  post: TPost;
  columns: number;
  gap: number;
};
GalleryGrid.Item = memo(
  ({ post, columns, gap }: TGaleryGridItemProps): JSX.Element => {
    const handlePress = useCallback(() => {
      // TODO - navigate to post detail screen
    }, []);
    const handleLongPress = useCallback(() => {
      // TODO - open simple post card
    }, []);

    return (
      <ImageContainer
        testID="GaleryGridItem"
        columns={columns}
        gap={gap}
        onPress={handlePress}
        onLongPress={handleLongPress}
      >
        <PostImage source={{ uri: post.medias[0].url }} />
        {post.medias.length > 1 ? <LayersIcon /> : null}
      </ImageContainer>
    );
  },
);

const ImageContainer = styled.Pressable<{ columns: number; gap: number }>`
  aspect-ratio: 1;
  width: ${({ columns, gap }) =>
    (SCREEN_WIDTH - gap * (columns - 1)) / columns}px;
  position: relative;
`;

const PostImage = styled.Image.attrs(() => ({
  resizeMode: 'cover',
}))`
  flex: 1;
`;

const LayersIcon = styled(LayersSvg).attrs(() => ({
  width: 16,
  height: 16,
}))`
  position: absolute;
  right: ${({ theme }) => theme.spacing.s};
  top: ${({ theme }) => theme.spacing.s};
  color: white;
  opacity: 0.9;
`;
