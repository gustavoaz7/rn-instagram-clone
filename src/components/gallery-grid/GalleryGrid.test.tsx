import React from 'react';
import { FlatList, Image } from 'react-native';
import { render } from '@testing-library/react-native';
import { GalleryGrid, TGalleryGridProps } from './GalleryGrid';
import { Providers } from '../../Providers';
import { SCREEN_WIDTH } from '../../utils/dimensions';
import { generateMockPost } from '../../data';
import LayersSvg from '../../../assets/svg/layers.svg';

describe('components - GalleryGrid', () => {
  const options = { wrapper: Providers };
  const defaultColumns = 3;
  const defaultGap = 2;
  const props: TGalleryGridProps = {
    posts: [...Array(4)].map(generateMockPost),
  };

  it('renders', () => {
    render(<GalleryGrid {...props} />, options);
  });

  it(`renders ${defaultColumns} columns with ${defaultGap} gap by default`, () => {
    const { UNSAFE_getByType, getAllByTestId } = render(
      <GalleryGrid {...props} />,
      options,
    );

    expect(UNSAFE_getByType(FlatList).props).toMatchObject({
      numColumns: defaultColumns,
      columnWrapperStyle: { paddingTop: defaultGap },
    });
    expect(getAllByTestId('GaleryGridItem')[0]).toHaveStyle({
      width:
        (SCREEN_WIDTH - defaultGap * (defaultColumns - 1)) / defaultColumns,
    });
  });

  it(`renders custom columns and gap`, () => {
    const columns = 4;
    const gap = 3;
    const { UNSAFE_getByType, getAllByTestId } = render(
      <GalleryGrid {...props} columns={columns} gap={gap} />,
      options,
    );

    expect(UNSAFE_getByType(FlatList).props).toMatchObject({
      numColumns: columns,
      columnWrapperStyle: { paddingTop: gap },
    });
    expect(getAllByTestId('GaleryGridItem')[0]).toHaveStyle({
      width: (SCREEN_WIDTH - gap * (columns - 1)) / columns,
    });
  });

  it('renders first media image', () => {
    const { UNSAFE_getAllByType } = render(<GalleryGrid {...props} />, options);

    UNSAFE_getAllByType(Image).forEach((image, i) => {
      expect(image.props.source.uri).toEqual(props.posts[i].medias[0].url);
    });
  });

  describe('when post has multiple medias ', () => {
    it('renders layers icon', () => {
      const posts = [generateMockPost({ mediasQty: 2 })];
      const { UNSAFE_getByType } = render(
        <GalleryGrid posts={posts} />,
        options,
      );

      expect(UNSAFE_getByType(LayersSvg)).toBeTruthy();
    });
  });

  describe('when post has a single media ', () => {
    it('doe snot render layers icon', () => {
      const posts = [generateMockPost({ mediasQty: 1 })];
      const { UNSAFE_queryByType } = render(
        <GalleryGrid posts={posts} />,
        options,
      );

      expect(UNSAFE_queryByType(LayersSvg)).toBeFalsy();
    });
  });
});
