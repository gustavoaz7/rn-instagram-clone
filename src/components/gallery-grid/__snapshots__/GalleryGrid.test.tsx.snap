// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - GalleryGrid matches snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "width": 750,
    }
  }
  data={
    Array [
      Object {
        "caption": "Tempora rerum minima dolor quibusdam blanditiis ea et. Earum iste iure expedita velit laudantium inventore quia qui. Distinctio consectetur accusamus voluptatum hic rem at quas. Incidunt ipsum sed illo molestiae vitae ullam sit consequatur autem. Ut voluptatem nisi. Velit ex dignissimos sed aut.",
        "createdAt": 1620488304326,
        "id": "0.7503052736155693",
        "location": "Daronville",
        "medias": Array [
          Object {
            "id": "0.9684701805514697",
            "owner": Object {
              "id": "0.9441732791858162",
              "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
              "username": "gustavoaz7",
            },
            "tappableObjects": Array [],
            "url": "http://placeimg.com/640/480/cats?0",
          },
        ],
        "owner": Object {
          "id": "0.5145090739885188",
          "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
          "username": "gustavoaz7",
        },
        "previewComments": Object {
          "comments": Array [
            Object {
              "createdAt": 1620488304326,
              "id": "0.8089725738800846",
              "owner": Object {
                "id": "0.3697436275449848",
                "profilePicUrl": "https://cdn.fakercloud.com/avatars/antonyzotov_128.jpg",
                "username": "Vivienne_Doyle",
              },
              "previewLikes": Object {
                "count": 1,
                "likes": Array [
                  Object {
                    "createdAt": 1620488304326,
                    "id": "0.7285493813636328",
                    "owner": Object {
                      "id": "0.44389326382814387",
                      "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
                      "username": "gustavoaz7",
                    },
                  },
                ],
              },
              "text": "Aspernatur quod ea unde repellendus magni nihil sunt.",
              "viewerHasLiked": false,
            },
          ],
          "count": 1,
        },
        "previewLikes": Object {
          "count": 3,
          "likes": Array [
            Object {
              "createdAt": 1620488304326,
              "id": "0.00955236152182315",
              "owner": Object {
                "id": "0.5939056183217004",
                "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
                "username": "gustavoaz7",
              },
            },
            Object {
              "createdAt": 1620488304326,
              "id": "0.3037633708355998",
              "owner": Object {
                "id": "0.15068375475313478",
                "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
                "username": "gustavoaz7",
              },
            },
            Object {
              "createdAt": 1620488304326,
              "id": "0.7916855374269904",
              "owner": Object {
                "id": "0.795496959059312",
                "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
                "username": "gustavoaz7",
              },
            },
          ],
        },
        "viewerHasLiked": false,
      },
      Object {
        "caption": "Tempora rerum minima dolor quibusdam blanditiis ea et. Earum iste iure expedita velit laudantium inventore quia qui. Distinctio consectetur accusamus voluptatum hic rem at quas. Incidunt ipsum sed illo molestiae vitae ullam sit consequatur autem. Ut voluptatem nisi. Velit ex dignissimos sed aut.",
        "createdAt": 1620488304326,
        "id": "0.41373594235159095",
        "location": "Daronville",
        "medias": Array [
          Object {
            "id": "0.08702038306262483",
            "owner": Object {
              "id": "0.33108232752437994",
              "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
              "username": "gustavoaz7",
            },
            "tappableObjects": Array [],
            "url": "http://placeimg.com/640/480/cats?0",
          },
        ],
        "owner": Object {
          "id": "0.6173003345303225",
          "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
          "username": "gustavoaz7",
        },
        "previewComments": Object {
          "comments": Array [
            Object {
              "createdAt": 1620488304326,
              "id": "0.4341613568949052",
              "owner": Object {
                "id": "0.9438394115623954",
                "profilePicUrl": "https://cdn.fakercloud.com/avatars/antonyzotov_128.jpg",
                "username": "Vivienne_Doyle",
              },
              "previewLikes": Object {
                "count": 1,
                "likes": Array [
                  Object {
                    "createdAt": 1620488304326,
                    "id": "0.8486227130149373",
                    "owner": Object {
                      "id": "0.04791428376514539",
                      "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
                      "username": "gustavoaz7",
                    },
                  },
                ],
              },
              "text": "Aspernatur quod ea unde repellendus magni nihil sunt.",
              "viewerHasLiked": false,
            },
          ],
          "count": 1,
        },
        "previewLikes": Object {
          "count": 3,
          "likes": Array [
            Object {
              "createdAt": 1620488304326,
              "id": "0.2126848050204142",
              "owner": Object {
                "id": "0.6099309914699516",
                "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
                "username": "gustavoaz7",
              },
            },
            Object {
              "createdAt": 1620488304326,
              "id": "0.3349606990400664",
              "owner": Object {
                "id": "0.34923798636800973",
                "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
                "username": "gustavoaz7",
              },
            },
            Object {
              "createdAt": 1620488304326,
              "id": "0.28787341966067115",
              "owner": Object {
                "id": "0.2815148051684657",
                "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
                "username": "gustavoaz7",
              },
            },
          ],
        },
        "viewerHasLiked": false,
      },
      Object {
        "caption": "Tempora rerum minima dolor quibusdam blanditiis ea et. Earum iste iure expedita velit laudantium inventore quia qui. Distinctio consectetur accusamus voluptatum hic rem at quas. Incidunt ipsum sed illo molestiae vitae ullam sit consequatur autem. Ut voluptatem nisi. Velit ex dignissimos sed aut.",
        "createdAt": 1620488304326,
        "id": "0.922939804031387",
        "location": "Daronville",
        "medias": Array [
          Object {
            "id": "0.3454974135594415",
            "owner": Object {
              "id": "0.9967745902879201",
              "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
              "username": "gustavoaz7",
            },
            "tappableObjects": Array [],
            "url": "http://placeimg.com/640/480/cats?0",
          },
        ],
        "owner": Object {
          "id": "0.9149912643433287",
          "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
          "username": "gustavoaz7",
        },
        "previewComments": Object {
          "comments": Array [
            Object {
              "createdAt": 1620488304326,
              "id": "0.10439887330194386",
              "owner": Object {
                "id": "0.3077320159906931",
                "profilePicUrl": "https://cdn.fakercloud.com/avatars/antonyzotov_128.jpg",
                "username": "Vivienne_Doyle",
              },
              "previewLikes": Object {
                "count": 1,
                "likes": Array [
                  Object {
                    "createdAt": 1620488304326,
                    "id": "0.7449996372561503",
                    "owner": Object {
                      "id": "0.1323044084147602",
                      "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
                      "username": "gustavoaz7",
                    },
                  },
                ],
              },
              "text": "Aspernatur quod ea unde repellendus magni nihil sunt.",
              "viewerHasLiked": false,
            },
          ],
          "count": 1,
        },
        "previewLikes": Object {
          "count": 3,
          "likes": Array [
            Object {
              "createdAt": 1620488304326,
              "id": "0.24024171888267531",
              "owner": Object {
                "id": "0.006463618503612434",
                "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
                "username": "gustavoaz7",
              },
            },
            Object {
              "createdAt": 1620488304326,
              "id": "0.03910877841727811",
              "owner": Object {
                "id": "0.11168779022959585",
                "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
                "username": "gustavoaz7",
              },
            },
            Object {
              "createdAt": 1620488304326,
              "id": "0.21778596999549849",
              "owner": Object {
                "id": "0.07886078925990381",
                "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
                "username": "gustavoaz7",
              },
            },
          ],
        },
        "viewerHasLiked": false,
      },
      Object {
        "caption": "Tempora rerum minima dolor quibusdam blanditiis ea et. Earum iste iure expedita velit laudantium inventore quia qui. Distinctio consectetur accusamus voluptatum hic rem at quas. Incidunt ipsum sed illo molestiae vitae ullam sit consequatur autem. Ut voluptatem nisi. Velit ex dignissimos sed aut.",
        "createdAt": 1620488304326,
        "id": "0.0369120343050815",
        "location": "Daronville",
        "medias": Array [
          Object {
            "id": "0.17852926881070164",
            "owner": Object {
              "id": "0.33047393738372",
              "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
              "username": "gustavoaz7",
            },
            "tappableObjects": Array [],
            "url": "http://placeimg.com/640/480/cats?0",
          },
        ],
        "owner": Object {
          "id": "0.1628838782770916",
          "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
          "username": "gustavoaz7",
        },
        "previewComments": Object {
          "comments": Array [
            Object {
              "createdAt": 1620488304326,
              "id": "0.7734841288575092",
              "owner": Object {
                "id": "0.3176454199925014",
                "profilePicUrl": "https://cdn.fakercloud.com/avatars/antonyzotov_128.jpg",
                "username": "Vivienne_Doyle",
              },
              "previewLikes": Object {
                "count": 1,
                "likes": Array [
                  Object {
                    "createdAt": 1620488304326,
                    "id": "0.18718871135554105",
                    "owner": Object {
                      "id": "0.9153662421447482",
                      "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
                      "username": "gustavoaz7",
                    },
                  },
                ],
              },
              "text": "Aspernatur quod ea unde repellendus magni nihil sunt.",
              "viewerHasLiked": false,
            },
          ],
          "count": 1,
        },
        "previewLikes": Object {
          "count": 3,
          "likes": Array [
            Object {
              "createdAt": 1620488304326,
              "id": "0.6086211068964358",
              "owner": Object {
                "id": "0.31525440508906266",
                "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
                "username": "gustavoaz7",
              },
            },
            Object {
              "createdAt": 1620488304326,
              "id": "0.14948593696820844",
              "owner": Object {
                "id": "0.7949024129460789",
                "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
                "username": "gustavoaz7",
              },
            },
            Object {
              "createdAt": 1620488304326,
              "id": "0.6198634447448861",
              "owner": Object {
                "id": "0.49257864606703805",
                "profilePicUrl": "https://avatars.githubusercontent.com/u/28453820?v=4",
                "username": "gustavoaz7",
              },
            },
          ],
        },
        "viewerHasLiked": false,
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "justifyContent": "space-between",
              "paddingTop": 2,
            },
          ]
        }
      >
        <View
          accessible={true}
          columns={3}
          focusable={true}
          gap={2}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "aspectRatio": 1,
                "position": "relative",
                "width": 248.66666666666666,
              },
            ]
          }
          testID="GaleryGridItem"
        >
          <Image
            resizeMode="cover"
            source={
              Object {
                "uri": "http://placeimg.com/640/480/cats?0",
              }
            }
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          />
        </View>
        <View
          accessible={true}
          columns={3}
          focusable={true}
          gap={2}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "aspectRatio": 1,
                "position": "relative",
                "width": 248.66666666666666,
              },
            ]
          }
          testID="GaleryGridItem"
        >
          <Image
            resizeMode="cover"
            source={
              Object {
                "uri": "http://placeimg.com/640/480/cats?0",
              }
            }
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          />
        </View>
        <View
          accessible={true}
          columns={3}
          focusable={true}
          gap={2}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "aspectRatio": 1,
                "position": "relative",
                "width": 248.66666666666666,
              },
            ]
          }
          testID="GaleryGridItem"
        >
          <Image
            resizeMode="cover"
            source={
              Object {
                "uri": "http://placeimg.com/640/480/cats?0",
              }
            }
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "justifyContent": "space-between",
              "paddingTop": 2,
            },
          ]
        }
      >
        <View
          accessible={true}
          columns={3}
          focusable={true}
          gap={2}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "aspectRatio": 1,
                "position": "relative",
                "width": 248.66666666666666,
              },
            ]
          }
          testID="GaleryGridItem"
        >
          <Image
            resizeMode="cover"
            source={
              Object {
                "uri": "http://placeimg.com/640/480/cats?0",
              }
            }
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
